compiler/src/codegen/compcore.rei

```diff
@@ -2,6 +2,7 @@ open Grain_typed;
 open Grain_middle_end;
 open Mashtree;
 open Wasm;
```

Comment:
We can probably drop this import now? In fact, do we need the original dependency at all?

Reply:
Meant to leave a comment about this.

We can't (just yet) because we still depend on the [Utf8 module](https://github.com/grain-lang/wasm-spec/blob/master/interpreter/binary/utf8.ml) that it provides for our strings (and it turns out there's no other module out there that does this exactly). I figured we'd have a discussion about breaking that functionality out into its own module, or potentially using WebAssembly's data section for strings (though tbd if this is possible right now).

Additionally, we currently write custom sections using the `Wasm` module, but it looks like we sprinkle a little ✨ Grain magic ✨ when we do that, so I wanted to touch that in another PR (rather than trying to change that out in this one).